File: r/bubble.r
bubble_sort<-function(arr){
  n<-length(arr)
  if(n<2){
    return(arr)
  }
  for(i in 1:(n-1)){
    s<- FALSE
    for(j in 1:(n-i)){
      if(arr[j]>arr[j+1]){
        t<-arr[j]
        arr[j]<-arr[j+1]
        arr[j+1]<-t
        s<-TRUE
      }
    }
    if (!s){break}
  }
  return(arr)
}

File: p/bubble.py
class bubble_sort():
    def __new__(cls,arr:list) -> list:
        for i in range(len(arr)-1):
            s=False
            for j in range(len(arr)-i-1):
                if arr[j]>arr[j+1]:
                    arr[j],arr[j+1]=arr[j+1],arr[j]
                    s=True
            if not s:
                break
        return arr
